src/main.rs

```diff
@@ -87,6 +87,30 @@ extern "C" {
 
 #[no_mangle]
 pub unsafe extern "C" fn stage_4() -> ! {
+    #[cfg(feature = "sse")]
+    {
+        use bit_field::BitField;
+        use x86_64::registers::control::Cr0;
+        let mut flags = Cr0::read_raw();
+        flags.set_bit(2, false);
+        flags.set_bit(1, true);
+        flags.set_bit(9, true);
+        flags.set_bit(10, true);
+        unsafe {
+            Cr0::write_raw(flags);
+        }
+        // For now, we must use inline ASM here
+        let mut cr4: u64;
+        unsafe {
+            asm!("mov %cr4, $0" : "=r" (cr4));
+        }
+        cr4.set_bit(9, true);
+        cr4.set_bit(10, true);
+        unsafe {
+            asm!("mov $0, %cr4" :: "r" (cr4) : "memory");
+        }
+    }
+
```

Comment:
Could you move this into an `enable_sse` function in a new `sse` module? Also, it should be called from `main`, not `stage_4`.